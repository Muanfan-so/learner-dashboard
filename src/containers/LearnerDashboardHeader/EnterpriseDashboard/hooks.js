import React from 'react';
import { hooks as appHooks } from 'data/redux';
import { StrictDict } from 'utils';
import * as module from './hooks';

export const state = StrictDict({
  showDialog: (val) => React.useState(val), // eslint-disable-line
  selectedItem: (val) => React.useState(val), // eslint-disable-line
});

export const useEnterpriseDashboardHook = () => {
  const { availableDashboards, mostRecentDashboard } = appHooks.useEnterpriseDashboardData();
  const [showDialog, setShowDialog] = module.state.showDialog(false);
  const [selectedItem, setSelectedItem] = module.state.selectedItem({});

  const beginSelectDashboardItem = (val) => () => {
    setSelectedItem(val);
    setShowDialog(true);
  };

  const cancelSelectDashboardItem = () => {
    setSelectedItem({});
    setShowDialog(false);
  };

  return {
    availableDashboards,
    mostRecentDashboard,
    showDialog,

    selectedItem,
    beginSelectDashboardItem,
    cancelSelectDashboardItem,
  };
};

export default useEnterpriseDashboardHook;
