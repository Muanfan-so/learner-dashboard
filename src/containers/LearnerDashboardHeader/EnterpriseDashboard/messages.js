import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  dashboard: {
    id: 'leanerDashboard.menu.dashboard.label',
    defaultMessage: 'Dashboard',
    description: 'The text for the user menu Dashboard navigation link.',
  },
  enterpriseDialogHeader: {
    id: 'leanerDashboard.enterpriseDialogHeader',
    defaultMessage: 'You have access to the {label} dashboard',
    description: 'title for enterpise dashboard dialog',
  },
  enterpriseDialogBody: {
    id: 'leanerDashboard.enterpriseDialogBody',
    defaultMessage: 'To access the coureses available to you through {label}, visit the {label} dashboard now.',
    description: 'Body text for enterpise dashboard dialog',
  },
  enterpriseDialogDismissButton: {
    id: 'leanerDashboard.enterpriseDialogDismissButton',
    defaultMessage: 'Dismiss',
    description: 'Dismiss button to cancel visiting dashboard',
  },
  enterpriseDialogConfirmButton: {
    id: 'leanerDashboard.enterpriseDialogConfirmButton',
    defaultMessage: 'Go To Dashboard',
    description: 'Confirm button to go to the dashboard url',
  },
});

export default messages;
