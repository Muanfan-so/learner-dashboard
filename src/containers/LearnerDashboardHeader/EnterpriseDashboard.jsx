import React from 'react';
// import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Dropdown, ModalDialog, ActionRow, Button } from '@edx/paragon';

import { hooks as appHooks } from 'data/redux';

import { nullMethod } from 'hooks';

import messages from './messages';

import { StrictDict } from 'utils';


export const state = StrictDict({
  showDialog: (val) => React.useState(val),
  selectedItem: (val) => React.useState(val),
});

export const useEnterpriseDashboardHook = () => {
  const { availableDashboards, mostRecentDashboard } = appHooks.useEnterpriseDashboardData();
  const [showDialog, setShowDialog] = state.showDialog(false);
  const [selectedItem, setSelectedItem] = state.selectedItem({});

  const beginSelectDashboardItem = (val) => () => {
    setSelectedItem(val);
    setShowDialog(true);
  }

  const cancelSelectDashboardItem = () => {
    setSelectedItem({});
    setShowDialog(false);
  }

  return {
    availableDashboards,
    mostRecentDashboard,
    showDialog,

    selectedItem,
    beginSelectDashboardItem,
    cancelSelectDashboardItem,
  }
}

export const EnterpriseDashboard = () => {
  const { formatMessage } = useIntl();
  const {
    availableDashboards,
    mostRecentDashboard,
    showDialog,

    selectedItem,
    beginSelectDashboardItem,
    cancelSelectDashboardItem
  } = useEnterpriseDashboardHook();

  return <>
    {
      availableDashboards.map(dashboard => (
        <Dropdown.Item onClick={beginSelectDashboardItem(dashboard)} active={dashboard.label === mostRecentDashboard.label} key={dashboard.label}>
          {dashboard.label} {formatMessage(messages.dashboard)}
        </Dropdown.Item>
      ))
    }
     <ModalDialog
      isOpen={showDialog}
      onClose={nullMethod}
      hasCloseButton={false}
      title=""
    >
      <div className="bg-white p-3 rounded shadow" style={{ textAlign: 'start' }}>
        <h4>{formatMessage(messages.enterpriseDialogHeader, { label: selectedItem.label })}</h4>
        <p>{formatMessage(messages.enterpriseDialogBody, { label: selectedItem.label })}</p>
        <ActionRow>
          <Button variant="tertiary" onClick={cancelSelectDashboardItem}>
            {formatMessage(messages.enterpriseDialogDismissButton)}
          </Button>
          <Button type="a" href={selectedItem.url}>{formatMessage(messages.enterpriseDialogConfirmButton)}</Button>
        </ActionRow>
      </div>
    </ModalDialog>
  </>
};

EnterpriseDashboard.propTypes = {};

export default EnterpriseDashboard;
