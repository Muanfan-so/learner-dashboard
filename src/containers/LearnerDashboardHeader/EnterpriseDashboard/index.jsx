import React from 'react';
// import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import {
  Dropdown, ModalDialog, ActionRow, Button,
} from '@edx/paragon';

import { nullMethod } from 'hooks';

import messages from './messages';
import useEnterpriseDashboardHook from './hooks';

export const EnterpriseDashboard = () => {
  const { formatMessage } = useIntl();
  const {
    availableDashboards,
    mostRecentDashboard,
    showDialog,

    selectedItem,
    beginSelectDashboardItem,
    cancelSelectDashboardItem,
  } = useEnterpriseDashboardHook();

  return (
    <>
      {availableDashboards.map((dashboard) => (
        <Dropdown.Item
          onClick={beginSelectDashboardItem(dashboard)}
          active={dashboard.label === mostRecentDashboard.label}
          key={dashboard.label}
        >
          {dashboard.label} {formatMessage(messages.dashboard)}
        </Dropdown.Item>
      ))}
      <ModalDialog
        isOpen={showDialog}
        onClose={nullMethod}
        hasCloseButton={false}
        title=""
      >
        <div
          className="bg-white p-3 rounded shadow"
          style={{ textAlign: 'start' }}
        >
          <h4>
            {formatMessage(messages.enterpriseDialogHeader, {
              label: selectedItem.label,
            })}
          </h4>
          <p>
            {formatMessage(messages.enterpriseDialogBody, {
              label: selectedItem.label,
            })}
          </p>
          <ActionRow>
            <Button variant="tertiary" onClick={cancelSelectDashboardItem}>
              {formatMessage(messages.enterpriseDialogDismissButton)}
            </Button>
            <Button type="a" href={selectedItem.url}>
              {formatMessage(messages.enterpriseDialogConfirmButton)}
            </Button>
          </ActionRow>
        </div>
      </ModalDialog>
    </>
  );
};

EnterpriseDashboard.propTypes = {};

export default EnterpriseDashboard;
