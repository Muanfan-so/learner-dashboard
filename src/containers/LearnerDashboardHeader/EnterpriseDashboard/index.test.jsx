import { shallow } from 'enzyme';
import EnterpriseDashboard from '.';

import useEnterpriseDashboardHook from './hooks';

jest.mock('./hooks', () => ({
  __esModule: true,
  default: jest.fn(),
}));

const enterpriseDashboardData = {
  availableDashboards: [
    { label: 'Personal', url: '/dashboard' },
    { label: 'edX, Inc.', url: '/edx-dashboard' },
    { label: 'Harvard', url: '/harvard-dashboard' },
  ],
  mostRecentDashboard: { label: 'edX, Inc.', url: '/edx-dashboard' },
};

describe('EnterpriseDashboard', () => {
  describe('snapshot', () => {
    const hookReturn = {
      ...enterpriseDashboardData,
      showDialog: false,

      selectedItem: {},
      beginSelectDashboardItem: jest.fn().mockName('beginSelectDashboardItem'),
      cancelSelectDashboardItem: jest
        .fn()
        .mockName('cancelSelectDashboardItem'),
    };

    test('initilized', () => {
      useEnterpriseDashboardHook.mockReturnValueOnce({ ...hookReturn });
      const el = shallow(<EnterpriseDashboard />);
      expect(el).toMatchSnapshot();
    });

    test('select item and open modal', () => {
      useEnterpriseDashboardHook.mockReturnValueOnce({
        ...hookReturn,
        selectedItem: enterpriseDashboardData.availableDashboards[0],
        showDialog: true,
      });
      const el = shallow(<EnterpriseDashboard />);
      expect(el).toMatchSnapshot();
    });
  });
});
