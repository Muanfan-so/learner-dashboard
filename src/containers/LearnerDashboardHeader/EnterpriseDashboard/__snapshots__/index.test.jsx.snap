// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterpriseDashboard snapshot initilized 1`] = `
<Fragment>
  <Dropdown.Item
    active={false}
    key="Personal"
  >
    Personal
     
    Dashboard
  </Dropdown.Item>
  <Dropdown.Item
    active={true}
    key="edX, Inc."
  >
    edX, Inc.
     
    Dashboard
  </Dropdown.Item>
  <Dropdown.Item
    active={false}
    key="Harvard"
  >
    Harvard
     
    Dashboard
  </Dropdown.Item>
  <ModalDialog
    hasCloseButton={false}
    isOpen={false}
    onClose={[MockFunction hooks.nullMethod]}
    title=""
  >
    <div
      className="bg-white p-3 rounded shadow"
      style={
        Object {
          "textAlign": "start",
        }
      }
    >
      <h4>
        You have access to the undefined dashboard
      </h4>
      <p>
        To access the coureses available to you through undefined, visit the undefined dashboard now.
      </p>
      <ActionRow>
        <Button
          onClick={[MockFunction cancelSelectDashboardItem]}
          variant="tertiary"
        >
          Dismiss
        </Button>
        <Button
          type="a"
        >
          Go To Dashboard
        </Button>
      </ActionRow>
    </div>
  </ModalDialog>
</Fragment>
`;

exports[`EnterpriseDashboard snapshot select item and open modal 1`] = `
<Fragment>
  <Dropdown.Item
    active={false}
    key="Personal"
  >
    Personal
     
    Dashboard
  </Dropdown.Item>
  <Dropdown.Item
    active={true}
    key="edX, Inc."
  >
    edX, Inc.
     
    Dashboard
  </Dropdown.Item>
  <Dropdown.Item
    active={false}
    key="Harvard"
  >
    Harvard
     
    Dashboard
  </Dropdown.Item>
  <ModalDialog
    hasCloseButton={false}
    isOpen={true}
    onClose={[MockFunction hooks.nullMethod]}
    title=""
  >
    <div
      className="bg-white p-3 rounded shadow"
      style={
        Object {
          "textAlign": "start",
        }
      }
    >
      <h4>
        You have access to the Personal dashboard
      </h4>
      <p>
        To access the coureses available to you through Personal, visit the Personal dashboard now.
      </p>
      <ActionRow>
        <Button
          onClick={[MockFunction cancelSelectDashboardItem]}
          variant="tertiary"
        >
          Dismiss
        </Button>
        <Button
          href="/dashboard"
          type="a"
        >
          Go To Dashboard
        </Button>
      </ActionRow>
    </div>
  </ModalDialog>
</Fragment>
`;
